Orange/widgets/data/oweditdomain.py

```diff
@@ -1555,6 +1565,18 @@ def get_data(self):
             tr.insert(0, self.format_cb.currentData())
         return var, tr
 
+    def _on_custom_change(self):
+        self.custom_text = self.custom_edit.text()
+        date_pat = r"%(-?)d|%(b|B)|%(-?)m|%(y|Y)|%(-?)j|%(-?)U|%(-?)W|%(a|A)|%w"
+        time_pat = r"%(-?)H|%(-?)I|%p|%(-?)M|%(-?)S|%f"
+        pattern1 = re.compile(date_pat)
+        pattern2 = re.compile(time_pat)
+        self.have_date = int(bool(pattern1.search(self.custom_text)))
+        self.have_time = int(bool(pattern2.search(self.custom_text)))
+        if self.format_cb.currentText() != "Custom format":
+            self.format_cb.setCurrentIndex(-1)
+        self.format_cb.emit()
```

Comment:
The proper call is `self.format_cb.currentIndexChanged.emit()`. But don't do this. It's better to call `self.variable_changed` directly, so the flow is more transparent. Also, `self.variable_changed` perhaps does something that shouldn't be done if the change is triggered by the user editing this line edit, in which case we'd split `self.variable_changed` in two functions and only call the second part here.)

Reply:
By the way, I think that you can't/shouldn't emit this signal. `setCurrentIndex` will already emit this signal (see https://doc.qt.io/qt-6/qcombobox.html#currentIndex-prop). That is, the proper code would be

```
    self.variable_changed()
```

but this will call `self.variable_changed()` twice (once because of the change in combo and once because of the change here), so you should probably write

```
if self.format_cb.currentText() != CUSTOM_FORMAT:
    self.format_cb.setCurrentIndex(self.format_cb.count() - 1)
else:
    self.variable_changed()
```